src/Handler/Customer/GenerateResetPasswordTokenHandler.php

```
@@ -30,10 +29,11 @@ public function __invoke(GenerateResetPasswordToken $generateResetPasswordToken)
 
         /** @var ShopUserInterface $user */
         $user = $this->userRepository->findOneByEmail($email);
-
-        Assert::notNull($user, sprintf('User with %s email has not been found.', $email));
```

The handler should throw an exception anyway, I suppose.

---

Where should this be caught? In the controller?

---

IMHO, we shouldn't catch them, because on application level it is a mistake. Validation of request should protect us from this scenario anyway

---

Yeah, makes sense.

---

I didn't get it! my logic is that the handler won't do any thing if the user email not found like sylius behaviour, The endpoint will return 204
`if (null === $user) {
            return;
        }`
Why It should throw exception?

---

True, but the this file is there for handling a command. This command COULD be called from the console where an error message might be helpful. So I agree with Lukasz that throwing an exception makes sense but you should definitely catch that and ignore it in the controller.

---

Okay I got it, I will add the commit